packages/grafana-runtime/src/services/TemplateSrv.ts

```diff
@@ -0,0 +1,24 @@
+import { TimeRange, ScopedVars, KeyValue } from '@grafana/ui';
+
+export interface TemplateSrv {
+  variables: any[];
+  init(variables: any[], timeRange?: TimeRange): void;
+  fillVariableValuesForUrl(params: KeyValue<string>, scopedVars?: ScopedVars): void;
+  highlightVariablesAsHtml(text?: string): string;
+  replaceWithText(target: string, scopedVars: ScopedVars): string;
+  replace(target: string, scopedVars?: ScopedVars, format?: string | Function): string;
```

Comment:
The react panels now access repace via [InterpolateFunction](https://github.com/grafana/grafana/blob/master/packages/grafana-ui/src/types/panel.ts#L7)  

I am not wild about the name (interpolate what?) but I don't think we need to expose all the details of this service -- replace vs replaceWithText?